pkgs/data/fonts/windows-fonts/default.nix

```diff
@@ -0,0 +1,101 @@
+{
+  lib,
+  stdenvNoCC,
+  fetchurl,
+  p7zip,
+}:
+let
+  windows-fonts =
+    {
+      pname,
+      version,
+      url,
+      sha256,
+      desc,
+    }:
+    let
+      languageFonts = import ./${pname}-languages.nix;
+      source =
+        with lib.attrsets;
+        mapAttrs' (
+          name: fonts:
+          nameValuePair (name) (builtins.concatStringsSep " " (map (font: "Windows/Fonts/" + font) fonts))
+        ) languageFonts;
+    in
+    stdenvNoCC.mkDerivation {
+      inherit pname version desc;
+
+      src = fetchurl { inherit url sha256; };
+
+      outputs = [ "out" ] ++ builtins.attrNames source;
+
+      nativeBuildInputs = [ p7zip ];
+
+      unpackPhase = ''
+        runHook preUnpack
+        7z x $src
+        7z x -aoa sources/install.wim Windows/{Fonts/"*".{ttf,ttc},System32/Licenses/neutral/"*"/"*"/license.rtf}
+        runHook postUnpack
+      '';
+
+      installPhase =
+        ''
+          runHook preInstall
+
+          mkdir -p $out/share/fonts/truetype
+        ''
+        + lib.strings.concatLines (
+          lib.lists.forEach (builtins.attrNames source) (
+            name:
+            (
+              ''install -Dm644 ${source.${name}} -t $''
+              + name
+              + ''
+                /share/fonts/truetype
+                              ln -s $''
+              + name
+              + ''
+                /share/fonts/truetype/*.{ttf,ttc} $out/share/fonts/truetype
+                              install -Dm644 Windows/System32/Licenses/neutral/*/*/license.rtf -t $''
+              + name
+              + ''
+                /share/licenses
+              ''
+            )
+          )
+        )
+        + ''
+          install -Dm644 Windows/System32/Licenses/neutral/*/*/license.rtf -t $out/share/licenses
+
+          runHook postInstall
+        '';
+
+      meta = with lib; {
+        description = "Some TrueType fonts from ${desc} (Arial, Bahnschrift, Calibri, Cambria, Candara, Consolas, Constantia...)";
+        homepage = "https://learn.microsoft.com/typography/";
+        license = licenses.unfree;
+        platforms = platforms.all;
+        maintainers = with maintainers; [ sobte ];
+
+        # Set a non-zero priority to allow easy overriding of the
+        # fontconfig configuration files.
+        priority = 5;
+      };
+    };
+in
+{
+  windows11 = windows-fonts {
+    pname = "windows11-fonts";
+    version = "10.0.22631.2428-2";
+    url = "https://software-static.download.prss.microsoft.com/dbazure/888969d5-f34g-4e03-ac9d-1f9786c66749/22631.2428.231001-0608.23H2_NI_RELEASE_SVC_REFRESH_CLIENTENTERPRISEEVAL_OEMRET_x64FRE_en-us.iso";
```

Comment:
What's the source of those?

Reply:
This pkgs attempts to download fonts directly from Microsoft, by
retrieving selective parts of the Windows 10 Enterprise 90-day evaluation edition.

visit: https://www.microsoft.com/en-us/evalcenter/evaluate-windows-10-enterprise
iso file: https://www.microsoft.com/en-us/evalcenter/download-windows-10-enterprise

win11 is the same as above

visit: https://www.microsoft.com/en-us/evalcenter/evaluate-windows-11-enterprise
iso file: https://www.microsoft.com/en-us/evalcenter/download-windows-11-enterprise